website/content/docs/concepts/oidc-provider.mdx

```diff
@@ -0,0 +1,144 @@
+
+* **Issuer URL**: used in the `iss` claim of ID tokens
+* **Allowed client IDs**: limits which clients can access the provider
+* **Scopes supported**: limits what identity information is available as claims.
+
+The issuer URL parameter is necessary for the validation of ID tokens by clients. If an URL parameter is not provided explicitly, it will default to a URL with Vault's [api_addr](/docs/configuration#api_addr) as the `scheme://host:port` component and `/v1/:namespace/identity/oidc/provider/:name` as the path component. This means tokens issued by a provider in a specified Vault cluster must be validated within that same cluster. If the issuer URL is provided explicitly, it must point to a Vault instance that is network-reachable by clients for ID token validation.
+
+The allowed client IDs parameter utilizes the list of client IDs that have been generated by Vault as a part of client registration. By default, all clients will be *disallowed*. Providing an asterisk(*) as the parameter value will allow all clients to use the provider. The scopes parameter employs a list of references to named scope resources. The values provided are discoverable by the `scopes_supported` key in the OIDC discovery document of the provider. By default, a provider will have the `openid` scope available. See the scopes section below for more details on the `openid` scope.
+
+### Scopes
+
+Providers may reference scope resources via the `scopes_supported` parameter to make specific identity information available as claims.
+
+A scope will have the following configuration parameters:
+
+* **Description**: identity information captured by the scope
+* **Template**: maps individual claims to Vault identity information
+
+
+The template parameter takes advantage of the [JSON-based templating](/api-docs/secret/identity/tokens#template) used by identity tokens for claims mapping. This means the parameter will take a JSON string of arbitrary structure where the values may be replaced with specific identity information. Template parameters that are not present for a Vault identity are omitted from the resulting claims without an error.
+
+
+Example of a JSON template for a scope:
+
+```
+{
+  "groups": {{identity.entity.groups.names}},
+  "email": {{identity.entity.metadata.user_email}},
+  "auth0": {
+      "name": {{identity.entity.aliases.$MNT_ACC.metadata.user}}
+  }
+}
+```
+
+The full list of template parameters include:
+
+* `identity.entity.id`
+* `identity.entity.name`
+* `identity.entity.groups.ids`
+* `identity.entity.groups.names`
+* `identity.entity.metadata`
+* `identity.entity.metadata.<metadata key>`
+* `identity.entity.aliases.<mount accessor>.id`
+* `identity.entity.aliases.<mount accessor>.name`
+* `identity.entity.aliases.<mount accessor>.metadata`
+* `identity.entity.aliases.<mount accessor>.metadata.<metadata key>`
+* `time.now`
+* `time.now.plus.<duration>`
+* `time.now.minus.<duration>`
```

Comment:
Maybe we should make this a table with the descriptions similar to https://www.vaultproject.io/docs/secrets/identity#token-contents-and-templates

Reply:
```suggestion
| Name                                                                             | Description                                                                             |
| :------------------------------------------------------------------------------- | :-------------------------------------------------------------------------------------- |
| `identity.entity.id`                                                             | The entity's ID                                                                         |
| `identity.entity.name`                                                           | The entity's name                                                                       |
| `identity.entity.groups.ids`                                                     | The IDs of the groups the entity is a member of                                         |
| `identity.entity.groups.names`                                                   | The names of the groups the entity is a member of                                       |
| `identity.entity.metadata`                                                       | Metadata associated with the entity                                                     |
| `identity.entity.metadata.<metadata key>`                                        | Metadata associated with the entity for the given key                                   |
| `identity.entity.aliases.<mount accessor>.id`                                    | Entity alias ID for the given mount                                                     |
| `identity.entity.aliases.<mount accessor>.name`                                  | Entity alias name for the given mount                                                   |
| `identity.entity.aliases.<mount accessor>.metadata`                              | Metadata associated with the alias for the given mount                                  |
| `identity.entity.aliases.<mount accessor>.metadata.<metadata key>`               | Metadata associated with the alias for the given mount and metadata key                 |
| `identity.entity.aliases.<mount accessor>.custom_metadata`                       | Custom metadata associated with the alias for the given mount                           |
| `identity.entity.aliases.<mount accessor>.custom_metadata.<custom_metadata key>` | Custom metadata associated with the alias for the given mount and custom metadata key   |
| `time.now`                                                                       | Current time as integral seconds since the Epoch                                        |
| `time.now.plus.<duration>`                                                       | Current time plus a Go-parsable [duration](https://golang.org/pkg/time/#ParseDuration)  |
| `time.now.minus.<duration>`                                                      | Current time minus a Go-parsable [duration](https://golang.org/pkg/time/#ParseDuration) |

```